### PR TITLE
Central Package Management Doc

### PR DESCRIPTION
**Pull request by JonDouglas**
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/2674

This is a work of progress to get an initial set of docs for the various features & getting started experience of central package management. 

Please suggest conceptual sections, changes and commit directly where you see fit so we can make the best doc possible.

### CODE REVIEW

File: docs/consume-packages/Central-Package-Management.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+description: Manage your dependencies in a central location and how you can get started with central package management.
+author: jondouglas
+ms.author: jodou
+ms.date: 2/25/2022
+ms.topic: conceptual
+---
+
+# Central Package Management
+
+Dependency management is a core feature of NuGet. Managing depdendencies for a single project can be easy. Managing dependencies for multi-project solutions can prove to be difficult as they start to scale in size and complexity. In situations where you manage common dependencies for many different projects, you can leverage NuGet's central package management features to do all of this from the ease of a single location.
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:12]:
```suggestion
Dependency management is a core feature of NuGet. Managing dependencies for a single project can be easy. Managing dependencies for multi-project solutions can prove to be difficult as they start to scale in size and complexity. In situations where you manage common dependencies for many different projects, you can leverage NuGet's central package management features to do all of this from the ease of a single location.
```

File: docs/consume-packages/Central-Package-Management.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+Historically, NuGet package dependencies have been managed in two main locations:
+
+- `packages.config` - An XML file used in older project types to maintain the list of packages referenced by the project.
+- `<PackageReference />` - An XML element used in new project types that manages NuGet dependencies directly within project files.
+
+Starting with [NuGet 7.0](..\release-notes\NuGet-7.0.md), you can centrally manage your dependencies in your projects with the addition of a `Directory.Packages.props` file.
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:19]: versioning; It's not really going to be nuget 7.0, but NuGet 6.4 that matches .NET 7.

File: docs/consume-packages/Central-Package-Management.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+The feature is available across all NuGet integrated tooling.
+
+* [Visual Studio 2022 and later](https://visualstudio.microsoft.com/downloads/)
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:23]: Update 4 probably.

File: docs/consume-packages/Central-Package-Management.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+* [.NET SDK 7.0.000 and later](https://dotnet.microsoft.com/download/dotnet/7.0)
+* [nuget.exe 7.0.0 and later](https://www.nuget.org/downloads)
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:25]: 6.4

File: docs/consume-packages/Central-Package-Management.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+Older tooling will ignore Central Package Management configurations and features. To use this feature to the fullest extent, ensure all your build environments use the latest compatible tooling versions.
+
+Central Package Management will apply to all project types – including .NET Framework – as long as compatible tooling is used.
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:29]: I'd probably say PackageReference projects - including old style csproj - while linking to the csproj.

File: docs/consume-packages/Central-Package-Management.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    <TargetFramework>net6.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" />
+  </ItemGroup>
+</Project>
+```
+
+Now you're using central package management and managing your versions in a central location!
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:60]: I think it'd be great if we can add a more detailed migration game, similar to what we did for source mapping. 

The rough steps would be: 

Run dotnet list packages or the equivalents. 
Start adding all top level dependencies in the Directory.Packages.props
Some guidance for resolving conflicts, including VersionOverride etc.
Remove *all* versions from PackageReference.

File: docs/consume-packages/Central-Package-Management.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+```
+
+- Project1 will evaluate the `Directory.Packages.props` file in the `Repository\Solution1\` directory.
+- Project2 will evaluate the `Directory.Packages.props` file in the `Repository\` directory.
+
+## Get started
+
+To fully onboard your repository, consider taking these steps:
+
+1. Create a new file at the root of your repository named `Directory.Packages.props` that declares your centrally defined package versions in.
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:87]: ignore my comment above :D The recommendation does apply though. 
Maybe provide some guidance about resolving conflicts, and advertise VersionOverride as a solution, but first encourage using the same version.

File: docs/consume-packages/Central-Package-Management.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+2. Declare `<PackageVersion />` items in your `Directory.Packages.props`.
+3. Declare `<PackageReference />` items without `Version` attributes in your project files.
+
+<!--For an idea of how central package management may look like, refer to our [samples repo](https://github.com/NuGet/Samples/tree/main/CentralPackageManagementExample).-->
+
+## Transitive pinning
+
+You can automatically override a transitive package version even without an explicit top-level `<PackageReference />` by opting into a feature known as transitive pinning. This promotes a transitive dependency to a top-level dependency implicitly on your behalf.
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:95]: > This promotes a transitive dependency to a top-level dependency implicitly on your behalf

This is likely to raise a lot questions. 
Maybe clarify it further that only where the transitive dependency exists, it'll be effectively act as being promoted top level. 

I don't have a great recommendation here. 
I'll comment again if I tihnk of something.

File: docs/consume-packages/Central-Package-Management.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+You can enable this feature by setting the MSBuild property `CentralPackageTransitivePinningEnabled` to `true` in a project or `Directory.Build.props` import file:
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:97]: Any reason why we shouldn't include Directory.Packages.props as an option?
In fact, I'd say it's maybe even the more appropriate place.

Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:97]: Any reason why we shouldn't include Directory.Packages.props as an option?
In fact, I'd say it's maybe even the more appropriate place.

File: docs/consume-packages/Central-Package-Management.md (hunk 2 above)
Comment (nkolev92) [reply docs/consume-packages/Central-Package-Management.md:19]: Given that we're adding the functionality in 6.2, any reason why we shouldn't announce the functionality in 6.2?

**Review by nkolev92 — COMMENTED** [review]
This looks great! Well done. 
Some suggestions for additions.

File: docs/consume-packages/Central-Package-Management.md (hunk 2 above)
Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:19]: Just placeholders. Will confirm with Jeff

File: docs/consume-packages/Central-Package-Management.md (hunk 5 above)
Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:29]: Good idea! Will fix.

File: docs/consume-packages/Central-Package-Management.md (hunk 7 above)
Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:87]: Good ideas, will add a couple more steps!

File: docs/consume-packages/Central-Package-Management.md (hunk 8 above)
Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:95]: Yeah, I tried to simplify what this means. I think an example could help, but also maybe we just wait until there's confusion on it and fix it then.

File: docs/consume-packages/Central-Package-Management.md (hunk 9 above)
Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:97]: Should we just include all three? I'm totally fine with adding that.

File: docs/consume-packages/Central-Package-Management.md (hunk 8 above)
Comment (nkolev92) [reply docs/consume-packages/Central-Package-Management.md:95]: Does calling it a pseudo top-level dependency make it better? 
That it clearly indicates there's more to it and it's an `always like this` thing.

File: docs/consume-packages/Central-Package-Management.md (hunk 9 above)
Comment (nkolev92) [reply docs/consume-packages/Central-Package-Management.md:97]: Yeah, I think that would be nice.

**Review by jeffkl — COMMENTED** [review]
Looks good, let's just nail down the release versions.  I imagine we'll also be updating this doc over time as people ask questions that it doesn't answer.

File: docs/consume-packages/Central-Package-Management.md (hunk 8 above)
Comment (nkolev92) [reply docs/consume-packages/Central-Package-Management.md:95]: Or `like` is just fine. 
The pseudo meaning in this case is the obvious one, `a pretend top-level dependency`.

Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:95]: My main worry is that it's hard enough for devs to distinguish top level & transitive today. Adding the word `pseudo` or `pretend` might confuse that further?

Comment (nkolev92) [reply docs/consume-packages/Central-Package-Management.md:95]: Fair, but misleading them could have a similar effect no? 

Maybe we change it to: 

> This promotes a transitive dependency to a top-level dependency implicitly on your behalf **when necessary**.

**Caps** is the addition.

Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:95]: yeah let's do that.

**Comment by JonDouglas** [comment]
@jeffkl Yep that's right.

File: docs/consume-packages/Central-Package-Management.md (hunk 5 above)
Comment (JonDouglas) [reply docs/consume-packages/Central-Package-Management.md:29]: Commit will come to fix :)

**Comment by JonDouglas** [comment]
Putting this up ready for review so we can publish it alongside VS 17.2 P3 if wanted :)

**Comment by nkolev92** [comment]
There's a few warnings raised by the build, in particular the link to the release notes, is that something we want to fix? Something comfortable leaving in for now?

File: docs/consume-packages/Central-Package-Management.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
+---
+title: Central Package Management
+description: Manage your dependencies in a central location and how you can get started with central package management.
+author: jondouglas
+ms.author: jodou
+ms.date: 2/25/2022
+ms.topic: conceptual
+---
+
+# Central Package Management
```
Comment (nkolev92) [inline docs/consume-packages/Central-Package-Management.md:10]: Awesome summary.

File: docs/consume-packages/Central-Package-Management.md (hunk 7 above)
Comment (nkolev92) [reply docs/consume-packages/Central-Package-Management.md:87]: Did you mean to add more steps here? 

We can always improve the doc later. Maybe we can enhance during docs days, as long as we have an issue to know what we need to do that.

**Review by nkolev92 — APPROVED** [review]

**Comment by JonDouglas** [comment]
Q: NuGet 6.2 won't likely go out for awhile, should we remove that for now or leave to 404 for now?

**Pull request merged**